components/PlantList.tsx

```diff
@@ -2,21 +2,79 @@ import React, { useEffect, useState } from 'react';
 import { getPlantSeasonality } from '@/api/supabase/queries/plantSeasonality';
 import { Plant } from '@/types/schema';
 
-export const PlantList = () => {
+interface PlantListProps {
+  harvest_season: string;
+  planting_type: string;
+  growing_season: string;
+}
+
+export const PlantList = (props: PlantListProps) => {
   const [plants, setPlants] = useState<Plant[]>([]);
+  const growingSeasonToMonth = new Map<string, string[]>([
+    ['Spring', ['MARCH', 'APRIL', 'MAY']],
+    ['Summer', ['JUNE', 'JULY', 'AUGUST']],
+    ['Fall', ['SEPTEMBER', 'OCTOBER', 'NOVEMBER']],
+    ['Winter', ['DECEMBER', 'JANUARY', 'FEBRUARY']],
+  ]);
 
   useEffect(() => {
     const fetchPlantSeasonality = async () => {
-      const plantList = await getPlantSeasonality('Tennessee');
+      // gets plants in Tennessee by default
+      const plantList = await getPlantSeasonality('TENNESSEE');
       setPlants(plantList);
     };
 
     fetchPlantSeasonality();
   }, []);
 
+  const checkGrowingSeason = (plant: Plant) => {
+    if (!props.growing_season) {
+      return true;
+    }
+
+    const months = growingSeasonToMonth.get(props.growing_season);
+
+    return (
+      months?.includes(plant.indoors_start) ||
+      months?.includes(plant.indoors_end) ||
+      months?.includes(plant.outdoors_start) ||
+      months?.includes(plant.outdoors_end)
```

Comment:
i'm wondering if we need to modify the logic here. what if the growing season is Summer, but the indoor time is April-Sept (i.e. it includes all Summer months, but the start and end times are not in the summer months), and outdoors is August-August? I don't think this example would realistically happen, but we also have to account for `LATE_MONTH` values which aren't included in the `growingSeasonToMonth` map. 
Perhaps we should try to all seasons that the growing season elapses and check if the inputted `growing_season` is in the list of elapsed seasons.
Perhaps using actual [enums](https://www.typescriptlang.org/docs/handbook/enums.html) might help here? (The type enums that we're using rn are just types and not actual enums oops)